Feature Engineering

Hwailaluta opened the pull request:
My variables were: exterior1st, exterior2nd, roofmatl, lotarea, lotfrontage, masvnrarea, BsmtFinSF1, BsmtFinSF2, and masvnrtype.

Hwailaluta committed:
Feature Engineering

My variables were: exterior1st, exterior2nd, roofmatl, lotarea, lotfrontage, masvnrarea, BsmtFinSF1, BsmtFinSF2, and masvnrtype.

srwight reviewed (commented):
Waiting to approve until we hear back from Kyle about duplicated effort

Hwailaluta committed:
Cleaned up

Changed from imputing with the mean to clipping. Added a doc string.

Hwailaluta commented on hend.py:
```diff
@@ -0,0 +1,56 @@
+    Q3 = df['LotFrontage'].quantile(.75)
+    IQR = Q3 - Q1
+
+    #I replaced outliers with the mean.
```
*Clipped, not mean.

srwight commented on hend.py:
```diff
@@ -0,0 +1,62 @@
+    pd.get_dummies(df['Exterior1st'])
+    pd.get_dummies(df['RoofMatl'])
+    pd.get_dummies(df['MasVnrType'])
+
```
get_dummies doesn't put them into the dataframe. It returns them. You'll need to capture them in a variable and add them back to your frame.

srwight commented on hend.py:
```diff
@@ -0,0 +1,62 @@
+    #df.drop(['BsmtFinSF1'], axis = 1, inplace=True)
+
+    features = df[['Exterior1st', 'RoofMatl', 'MasVnrType', 'MasVnrArea', 'LotFrontage', 'LotArea']]
+
```
Your return df will have other columns, too, which will be the dummies you made up above.

srwight commented on hend.py:
```diff
@@ -0,0 +1,62 @@
+
+    #Due to the presence of 'TotalBsmtSF' (which is a sum of the below two variables) and sparseness of data, I drop these two.
+    #df.drop(['BsmtFinSF1'], axis = 1, inplace=True)
+    #df.drop(['BsmtFinSF1'], axis = 1, inplace=True)
```
Is this a copy/paste error? you have the same line twice, which will throw an error when the second line can't find the column.

srwight commented on hend.py:
```diff
@@ -30,17 +23,13 @@ def feature_extract(df):
-    for entry in df['LotFrontage']:
-        if entry > (Q3 + 1.5 * IQR):
-            df['LotFrontage'].replace(entry, df['LotFrontage'].mean(), inplace = True)
+    df['LotFrontage'].clip(0,((Q3 + 1.5 * IQR))
```
You have a hanging paren '(' after clip that isn't resolved. It looks like your goal is to clip the data to either zero or the 3rd quartile plus 1.5 IQR's, right? That looks good. But you might want to drop one of the '(' before Q3. I think that will solve your problem.

srwight commented on hend.py:
```diff
@@ -30,17 +23,13 @@ def feature_extract(df):
-    for entry in df['MasVnrArea']:
-    if entry > (Q3 + 1.5 * IQR):
-        df['MasVnrArea'].replace(entry, df['MasVnrArea'].mean(), inplace = True)
+    df['MasVnrArea'].clip(0,((Q3 + 1.5 * IQR))
```
Same issue as above

srwight committed:
Update hend.py

srwight commented on hend.py:
```diff
@@ -0,0 +1,56 @@
+
+def main():
+    df=pd.read_csv('train.csv')
+    print(feature_extract(df))
```
you still had 'engineer()' here. I swapped it out.

srwight reviewed (changes requested):
See comments within - you have some syntax issues, and your feature variables are not getting captured.

Hwailaluta committed:
Fixed dummies and syntax

Hwailaluta committed:
Removed redundancy

I had taken over Kyle's data previously as he was unable to take up the work at that time (Christmas break). I've removed the redundant work and cleaned the code some more.

srwight reviewed (approved):
Yup
